packages/flutter/lib/src/cupertino/magnifier.dart

```diff
@@ -245,29 +245,60 @@ class CupertinoMagnifier extends StatelessWidget {
         color: Color.fromARGB(25, 0, 0, 0),
         blurRadius: 11,
         spreadRadius: 0.2,
+        blurStyle: BlurStyle.outer,
       ),
     ],
+    this.clipBehavior = Clip.none,
     this.borderSide =
         const BorderSide(color: Color.fromARGB(255, 232, 232, 232)),
     this.inOutAnimation,
   });
 
-  /// The shadows displayed under the magnifier.
+  /// A list of shadows cast by the [Magnifier].
+  ///
+  /// If the shadows use a [BlurStyle] that paints inside the shape, or if they
+  /// are offset, then a [clipBehavior] that enables clipping (such as
+  /// [Clip.hardEdge]) is recommended, otherwise the shadow will occlude the
+  /// magnifier (the shadow is drawn above the magnifier so as to not be
+  /// included in the magnified image).
+  ///
+  /// A shadow that uses [BlurStyle.outer] and is not offset does not need
+  /// clipping.
+  ///
+  /// By default, the [shadows] are not offset and use [BlurStyle.outer], and
+  /// correspondingly the default [clipBehavior] is [Clip.none].
   final List<BoxShadow> shadows;
 
+  /// Whether and how to clip the [shadows] that render inside the loupe.
+  ///
+  /// Defaults to [Clip.none], which is useful if the shadow will not paint
+  /// where the magnified image appears, or if doing so is intentional (e.g. to
+  /// blur the edges of the magnified image).
+  ///
+  /// The default configuration of [CupertinoMagnifier] does not render inside
+  /// the loupe (the shadows are not offset and use [BlurStyle.outer]).
+  ///
+  /// Other values (e.g. [Clip.hardEdge]) are recommended when the [shadows]
+  /// have an offset.
+  ///
+  /// See the discussion at [shadows].
```

Comment:
Nit: Would this be any better as a "See also"?

Reply:
I tried both ways... I sort of felt like "see also" made it seem less immediately relevant, more of a "if you're browsing, you may also like these things" as opposed to "there's more to read on this right now, over there".

Reply:
Sounds good to me!